src/sw/redis++/redlock.hpp

```diff
@@ -0,0 +1,103 @@
+/**************************************************************************
+   Copyright (c) 2019
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ *************************************************************************/
+
+#ifndef REDISPLUSPLUS_REDLOCK_H
+#define REDISPLUSPLUS_REDLOCK_H
+
+#include "redis++.h"
+#include <unistd.h>
+#include <openssl/rc4.h>
+
+namespace sw {
+
+namespace redis {
+
+class RandomBufferInterface
+{
+public:
+	virtual ~RandomBufferInterface() {}
+	virtual std::string getUpdatedString() = 0;
+};
+
+template <typename RedisInstance>
+class Redlock
+{
+public:
+	// In the constructor, we load teh scripts and initialize their
+	// SHA1s so that we can check if the script exists, before calling them.
+	Redlock(RedisInstance& instance, RandomBufferInterface& randomBuffer) :
+		_randomBuffer(randomBuffer),
+		_instance(instance),
+		_unlockScript("\
```

Comment:
You can try a new C++ feature: [raw string literal](https://en.cppreference.com/w/cpp/language/string_literal)

Reply:
Do you mean `_unlockScript(R"...")`?
If so, I could not get this right (lack of C++ skills), so could you please show me how to do it in `RedLock::RedLock()` in `redlock.cpp` of PR #33?

Reply:
You can try this: * _unlockScript(R"(raw_characters )")*